examples/hpc-c/hello-newlib-hpc/CMakeLists.txt

```diff
@@ -0,0 +1,38 @@
+cmake_minimum_required(VERSION 3.10)
+
+# Set riscv target
+set(CMAKE_SYSTEM_NAME Generic)
+set(CMAKE_SYSTEM_PROCESSOR riscv64)
+
+project(hello-newlib-hpc C)
+
+# Add C-runtime 0
+set(CRT0 ${CMAKE_CURRENT_SOURCE_DIR})
+set(INC_PATH "/opt/headsail-newlib/cygwin-newlib/riscv64-unknown-elf/include")
+set(LIB_PATH "/opt/headsail-newlib/cygwin-newlib/riscv64-unknown-elf/lib")
```

Comment:
First I would reduce this to one external variable`$NEWLIB_PATH` since it looks like `include` and `lib` are placed consistently.

```cmake
set(INC_PATH "${NEWLIB_PATH}/include")
set(LIB_PATH "${NEWLIB_PATH}/lib")
```

Then the path to Newlib can be set found with `find_path`and `CMAKE_PREFIX_PATH` something like this:

```cmake
list(APPEND CMAKE_PREFIX_PATH "/opt/headsail-newlib" "~/work/newlib-build/build")

if(NOT NEWLIB_PATH)
  find_path(
    NEWLIB_PATH
    NAMES "include/" "lib/"
    PATH_SUFFIXES "build/riscv64-unknown-elf" "riscv64-unknown-elf"
  )  
  if(NOT NEWLIB_PATH)
    message(FATAL_ERROR "Newlib not found")
  endif()
endif()

set(INC_PATH "${NEWLIB_PATH}/include")
set(LIB_PATH "${NEWLIB_PATH}/lib")
```

Now CMAKE can be given the path directly with `cmake .. -DNEWLIB_PATH=path_to_newlib` and if that is not given CMAKE tries to search from default locations.